lint/registry.go

```diff
@@ -0,0 +1,91 @@
+package lint
+
+import (
+	"fmt"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+// MakeRegistryFromAllFiles creates a *protoregistry.Files with all dependencies resolved, provided that
+// any import in any FileDescriptorProto in descs is also contained in descs. Missing dependencies will
+// be filled with placeholders
+//
+// In other words, if for any i, descs[i] imports "a.proto", then the FileDescriptorProto for "a.proto"
+// must also be present in files. If a dependency is missing from descs,
+func MakeRegistryFromAllFiles(descs []*descriptorpb.FileDescriptorProto) (*protoregistry.Files, error) {
+	filesSet, err := makeFilesSet(descs)
+	if err != nil {
+		return nil, err
+	}
+
+	f := files{
+		reg:      new(protoregistry.Files),
+		filesSet: filesSet,
+	}
+
+	for _, desc := range descs {
+		if err := f.register(desc.GetName()); err != nil {
+			return nil, err
+		}
+	}
+
+	return f.reg, nil
+}
+
+type files struct {
+	reg *protoregistry.Files
+
+	filesSet map[string]*entry
+}
+
+type entry struct {
+	descProto   *descriptorpb.FileDescriptorProto
+	desc        protoreflect.FileDescriptor
+	registering bool
```

Comment:
`visited` is a common word in graph search, which will help others understand the algorithm here.

Reply:
`visted` does not accurately describe what's happening. It's past tense, which will be confusing, because after a given file is registered it will be set to false. `visiting` is more accurate, but the function in named `register`, so this more accurately describes "the algorithm is currently in the `register` function for this `entry`.